code/modules/materials/Mat_FabRecipes.dm

```diff
@@ -525,6 +525,33 @@
 			newObj.set_loc(getOutputLocation(owner))
 		return
 
+/datum/matfab_recipe/shoes
+	name = "Shoes"
+	desc = "A custom pair of shoes."
+	category = "Clothing"
+
+	New()
+		required_parts.Add(new/datum/matfab_part/clothororganicorrubber {part_name = "Upper"; required_amount = 2} ())
+		required_parts.Add(new/datum/matfab_part/anymat {part_name = "Sole"; required_amount = 2} ())
+		required_parts.Add(new/datum/matfab_part/optionalanymat {part_name = "Optional Toe Tip"; required_amount = 2} ())
+		..()
+
+	build(amount, var/obj/machinery/nanofab/owner)
+		for(var/i=0, i<amount, i++)
+			var/obj/item/clothing/shoes/crafted/newObj = new()
+			var/obj/item/upper = getObjectByPartName("Upper")
+			var/obj/item/sole = getObjectByPartName("Sole")
+			var/obj/item/toe = getObjectByPartName("Optional Toe Tip")
+			if(toe && toe.material)
+				newObj.setMaterial(toe.material)
+				newObj.desc = "[toe.material.name]-tipped [upper.material.name] shoes. The soles are made of [sole.material.name]."
```

Comment:
maybe, `foo-toed` instead of `foo-tipped`? Like, steel-toed boots? tipped makes it sound like you stapeld an arrowhead on the front.

Reply:
ah yea